composer/loggers/mlflow_logger.py

```diff
@@ -39,67 +44,100 @@ def __init__(
     ) -> None:
         try:
             import mlflow
+            from mlflow import MlflowClient
+            from mlflow.utils.autologging_utils import MlflowAutologgingQueueingClient
         except ImportError as e:
             raise MissingConditionalImportError(extra_deps_group='mlflow',
                                                 conda_package='mlflow',
                                                 conda_channel='conda-forge') from e
-        del mlflow
         self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
 
         self.run_name = run_name
         self.experiment_name = experiment_name
         self._rank_zero_only = rank_zero_only
-        self.tracking_uri = tracking_uri
+        self.tracking_uri = str(tracking_uri or mlflow.get_tracking_uri())
+        self._last_flush_time = time.time()
+        self._flush_interval = flush_interval
+        if self._enabled:
+            # Set up MLflow state
+            self._run_id = None
+            if self.experiment_name is None:
+                self.experiment_name = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.name,
+                                                 DEFAULT_MLFLOW_EXPERIMENT_NAME)
+            self._mlflow_client = MlflowClient(self.tracking_uri)
+            # Create an instance of MlflowAutologgingQueueingClient - an optimized version
+            # of MlflowClient - that automatically batches metrics together and supports
+            # asynchronous logging for improved performance
+            self._optimized_mlflow_client = MlflowAutologgingQueueingClient(self.tracking_uri)
+            # set experiment. we use MlflowClient for experiment retrieval and creation
+            # because MlflowAutologgingQueueingClient doesn't support it
+            env_exp_id = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_ID.name, None)
+            if env_exp_id is not None:
+                self._experiment_id = env_exp_id
+            else:
+                exp_from_name = self._mlflow_client.get_experiment_by_name(name=self.experiment_name)
+                if exp_from_name is not None:
+                    self._experiment_id = exp_from_name.experiment_id
+                else:
+                    self._experiment_id = (self._mlflow_client.create_experiment(name=self.experiment_name))
+        del mlflow
 
     def init(self, state: State, logger: Logger) -> None:
         import mlflow
         del logger  # unused
 
-        if self.experiment_name is None:
-            self.experiment_name = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.name,
-                                             DEFAULT_MLFLOW_EXPERIMENT_NAME)
-
         if self.run_name is None:
             self.run_name = state.run_name
 
         # Adjust name and group based on `rank_zero_only`.
         if not self._rank_zero_only:
             self.run_name += f'-rank{dist.get_global_rank()}'
 
+        # start run
```

Comment:
```suggestion
        # Start run
```
nit (which is really a nit from the previous code :P)